src/worker/predictive-text-studio-worker-impl.ts

```diff
@@ -20,17 +22,26 @@ function doNothing() {
 
 export class PredictiveTextStudioWorkerImpl
   implements PredictiveTextStudioWorker {
-  private storage: Storage;
-
-  constructor(storage = new Storage()) {
-    this.storage = storage;
+  constructor(
+    private storage = new Storage(),
+    private keymanAPI = new KeymanAPI()
+  ) {
+    this.getLanguageData();
   }
 
   async updateBCP47Tag(bcp47Tag: string): Promise<void> {
     await this.storage.updateBCP47Tag(bcp47Tag);
     return this.generateKMPFromStorage();
   }
 
+  async getLanguageData(): Promise<void> {
+    this.keymanAPI.fetchLanaguageData().then((languages: KeyboardData[]) => {
+      languages.forEach(async (data) => {
+        await this.storage.addKeyboardData(data.langauge, data.bcp47Tag);
+      });
+    });
```

Comment:
Every time the worker starts up, it will fetch the entire language list. This is bad because it should really only be fetched maybe once a day, at most. But if you open a new tab, a new worker thread starts, which issues this request to the Keyman API 💀 

Is there a way you can set a limit to how often the data is fetched?